os-audit/cos-auditd-logging.yaml

```diff
@@ -0,0 +1,172 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cos-auditd
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: cos-auditd-logging
+  namespace: cos-auditd
+  annotations:
+    kubernetes.io/description: 'DaemonSet that enables Linux auditd logging on COS nodes.'
+spec:
+  selector:
+    matchLabels:
+      name: cos-auditd-logging
+  template:
+    metadata:
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ""
+      labels:
+        name: cos-auditd-logging
+    spec:
+      nodeSelector:
+        cloud.google.com/gke-os-distribution: cos
+      volumes:
+      - hostPath:
+          path: /
+        name: host
+      - hostPath:
+          path: /var/log
+        name: varlog
+      - hostPath:
+          path: /usr/lib64
+        name: libsystemddir
+      - configMap:
+          defaultMode: 420
+          name: fluentd-gcp-config-cos-auditd
+        name: config-volume
+      initContainers:
+      - name: cos-auditd-setup
+        image: ubuntu
+        command: ["chroot", "/host", "systemctl", "start", "cloud-audit-setup"]
+        volumeMounts:
+        - name: host
+          mountPath: /host
+        resources:
+          requests:
+            memory: "10Mi"
+            cpu: "10m"
+      containers:
+      - command:
+        - /bin/sh
+        - -c
+        - /run.sh $FLUENTD_ARGS 2>&1 >>/var/log/fluentd-audit.log
```

Comment:
Why write the logs to a file, rather than just using the standard log handling? (e.g. kubectl logs doesn't work with this approach)

Reply:
I agree that we should use container logging as otherwise users have to ssh onto the nodes to get the logs. But I'm not sure this is done with intention. How about I creating an issue to discuss this later?

Reply:
Tracked in https://github.com/GoogleCloudPlatform/k8s-node-tools/issues/2